mysql-test/suite/galera/t/galera_sst_xtrabackup-v2_super_read_only.test

```diff
@@ -1,26 +1,97 @@
+# Test 1: Test that the cluster starts up
+# --------------------------------
+
+--connection node_2
+--echo [connection node_2]
+
+--let $NODE2_CONFIG_FILE = $MYSQLTEST_VARDIR/tmp/galera_sst_xtrabackup-v2_super_read_only.test.cnf
+--copy_file $MYSQLTEST_VARDIR/my.cnf $NODE2_CONFIG_FILE
+--append_file $NODE2_CONFIG_FILE
+	[mysqld]
+	super_read_only=ON
+EOF
+
+# --------------------------------
+# Test 1a: Restart node2 (no sst)
+# --------------------------------
+
+# restart the server with super-read-only=ON (no SST)
+--echo "# restarting node2 (no SST)"
+--let $restart_hide_args=1
+--let $restart_parameters = "restart: --defaults-file=$NODE2_CONFIG_FILE"
+--source include/restart_mysqld.inc
+--source include/wait_until_connected_again.inc
+
+SET SESSION wsrep_sync_wait = 0;
 --let $wait_condition = SELECT VARIABLE_VALUE = 2 FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_cluster_size';
 --source include/wait_condition.inc
 
---connection node_2
+SELECT 1;
+
+# We should not find the super-read-only warnings in the logs
+--let $assert_text = Could not find warnings about super-read-only
+--let $assert_select = super-read-only
+--let $assert_count = 0
+--let $assert_file = $MYSQLTEST_VARDIR/log/mysqld.2.err
+--let $assert_only_after = CURRENT_TEST
+--source include/assert_grep.inc
+
+
+# --------------------------------
+# Test 1b: Restart node2 (with sst)
+# --------------------------------
+--echo "# shutting down node2"
 SET SESSION wsrep_sync_wait = 0;
 --source include/shutdown_mysqld.inc
+--source include/wait_until_disconnected.inc
+
+# remove the grastate.dat file to force an SST
 --remove_file $MYSQLTEST_VARDIR/mysqld.2/data/grastate.dat
---exec sed -i '/^\[mysqld.2\]/a super_read_only=ON' $MYSQLTEST_VARDIR/my.cnf
+
+# restart the server with super-read-only=ON (SST)
+--echo "# restarting node2 (with SST)"
+--let $restart_hide_args=1
+--let $start_mysqld_params = "--defaults-file=$NODE2_CONFIG_FILE"
 --let $_expect_file_name= $MYSQLTEST_VARDIR/tmp/mysqld.2.expect
 --source include/start_mysqld.inc
+--source include/wait_until_connected_again.inc
+
+SET SESSION wsrep_sync_wait = 0;
+--let $wait_condition = SELECT VARIABLE_VALUE = 2 FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+--source include/wait_condition.inc
+
+SELECT 1;
+
+# We should not find the super-read-only warnings in the logs
+--let $assert_text = Could not find warnings about super-read-only
+--let $assert_select = super-read-only
+--let $assert_count = 0
+--let $assert_file = $MYSQLTEST_VARDIR/log/mysqld.2.err
+--let $assert_only_after = CURRENT_TEST
+--source include/assert_grep.inc
+
 
 #cleanup
 --connection node_2
-SET GLOBAL super_read_only = OFF;
-SET GLOBAL read_only = OFF;
---exec sed -i '/^super_read_only=ON/d' $MYSQLTEST_VARDIR/my.cnf
+--echo [connection node_2]
+
+--let $restart_parameters = "restart:"
+--source include/restart_mysqld.inc
+--source include/wait_until_connected_again.inc
```

Comment:
This is redundant. We can avoid including `wait_until_connected_again.inc` after using either the `start_mysqld.inc` and `restart_mysqld.inc`, as the same check is already done inisde the inc files.